Woo Express: Add plan features and feature groups

markbiek opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to #75383

## Proposed Changes

This adds all of the Woo Express-specific plan features and feature groups.

This is part of several PRs related to plan features and the plan comparison grid:
* #75918 (this PR) which adds all of the plan features and groups.
* #75769 which enables the plan comparison grid for Woo Express.
* #75883 which adds footnotes to the plan comparison grid.

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

This Slack conversation (p1681499986012049-slack-C0Q664T29) has a link to a spreadsheet with all of the features.

 * Inspect the code and verify that the correct features exist and are listed under the correct groups.


## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

markbiek committed:
Add new Woo Express features

markbiek committed:
Add new Woo Express feature groups

markbiek committed:
Add 50GB to list of available storage strings

markbiek committed:
Remove min/max qty from Essential plan features

sixhours commented on client/lib/plans/features-list.tsx:
```diff
@@ -2168,6 +2194,139 @@ export const FEATURES_LIST: FeatureList = {
+	/* START: Woo Express Features */
+	[ FEATURE_WOOCOMMERCE_STORE ]: {
+		getSlug: () => FEATURE_WOOCOMMERCE_STORE,
+		getTitle: () => 'WooCommerce store',
```
Do all these `getTitle` values need to be wrapped in translate functions? I see some are and some are not.

markbiek replied:
They definitely should be wrapped in translate functions. Thanks!

sixhours reviewed (approved):
I think this is good! I went through line by line and compared with the spreadsheet (although there are a *lot* of things to compare so I worry there's a lot of room for human error on my part -- I wonder if it makes sense to write some tests to ensure we're including the right features with the right plans/groups? Or does it matter so much, since these are constants and we're not manipulating the data very much?)

I noticed the spreadsheet has a typo:
<img width="1107" alt="Screen Shot 2023-04-18 at 3 11 08 PM" src="https://user-images.githubusercontent.com/2124984/232883007-031a3f54-8882-4107-bebd-f0bb17a50184.png">

markbiek commented:
>  I wonder if it makes sense to write some tests to ensure we're including the right features with the right plans/groups?

We don't seem to have tests for any of the other constants but this is probably as a good a time as any to add them!

markbiek committed:
Make sure all new feature titles are translated

markbiek committed:
Add missing recurring payments feature

The Small and Medium Woo Express plans were missing FEATURE_RECURRING_PAYMENTS as a comparison grid feature

markbiek commented:
Screenshot for translation:

![image](https://user-images.githubusercontent.com/917632/233059791-d7eff5cf-dda1-45ad-88bc-bc9d37d72910.png)
